plaso/parsers/azure_applicationgatewayaccess.py

```diff
@@ -0,0 +1,252 @@
+        event_data.client_ip = properties_json_log_entry.get('clientIP')
+        event_data.client_port = properties_json_log_entry.get('clientPort')
+        event_data.http_method = properties_json_log_entry.get('httpMethod')
+        event_data.original_request_uri_with_args = (
+          properties_json_log_entry.get('originalRequestUriWithArgs')
+        )
+        event_data.request_uri = properties_json_log_entry.get('requestUri')
+        event_data.request_query = properties_json_log_entry.get('requestQuery')
+        event_data.user_agent = properties_json_log_entry.get('userAgent')
+        event_data.http_status = properties_json_log_entry.get('httpStatus')
+        event_data.http_version = properties_json_log_entry.get('httpVersion')
+        event_data.received_bytes = properties_json_log_entry.get(
+          'receivedBytes'
+        )
+        event_data.sent_bytes = properties_json_log_entry.get('sentBytes')
+        event_data.client_response_time = properties_json_log_entry.get(
+          'clientResponseTime'
+        )
+        event_data.time_taken = properties_json_log_entry.get('timeTaken')
+        event_data.waf_evaluation_time = properties_json_log_entry.get(
+          'WAFEvaluationTime'
+        )
+        event_data.waf_mode = properties_json_log_entry.get('WAFMode')
+        event_data.transaction_id = properties_json_log_entry.get(
+          'transactionId'
+        )
+        event_data.ssl_enabled = properties_json_log_entry.get('sslEnabled')
+        event_data.ssl_cipher = properties_json_log_entry.get('sslCipher')
+        event_data.ssl_protocol = properties_json_log_entry.get('sslProtocol')
+        event_data.ssl_client_verify = properties_json_log_entry.get(
+          'sslClientVerify'
+        )
+        event_data.ssl_client_certificate_fingerprint = (
+          properties_json_log_entry.get('sslClientCertificateFingerprint')
+        )
+        event_data.ssl_client_certificate_issuer_name = (
+          properties_json_log_entry.get('sslClientCertificateIssuerName')
+        )
+        event_data.server_routed = properties_json_log_entry.get('serverRouted')
+        event_data.server_status = properties_json_log_entry.get('serverStatus')
+        event_data.server_response_latency = properties_json_log_entry.get(
+          'serverResponseLatency'
+        )
+        event_data.original_host = properties_json_log_entry.get('originalHost')
+        event_data.host = properties_json_log_entry.get('host')
+      else:
+        parser_mediator.ProduceExtractionWarning(
+          'unable to parse application gateway access log event:'
+          'missing properties field'
+        )
+        continue
+
+      try:
+        date_time = dfdatetime_time_elements.TimeElementsInMicroseconds()
+        date_time.CopyFromStringISO8601(time_string)
+      except ValueError as exception:
+        parser_mediator.ProduceExtractionWarning(
+          f'Unable to parse time string: {time_string} with error: '
```

Comment:
style guide nit: four spaces for hanging indents here and the line below (ref: https://github.com/log2timeline/l2tdocs/blob/main/process/Style-guide.md#indentation)